src/library-authoring/data/api.ts

```diff
@@ -36,3 +45,15 @@ export async function getContentLibrary(libraryId?: string): Promise<ContentLibr
   const { data } = await getAuthenticatedHttpClient().get(getContentLibraryApiUrl(libraryId));
   return camelCaseObject(data);
 }
+
+/**
+ * Fetch block types of a library
+ */
+export async function getLibraryBlockTypes(libraryId?: string): Promise<LibraryBlockType[]> {
+  if (!libraryId) {
+    throw new Error('libraryId is required');
+  }
+
```

Comment:
Why are you making `libraryId` an optional parameter in TypeScript then throwing a runtime error? Just require it.

```suggestion
export async function getLibraryBlockTypes(libraryId: string): Promise<LibraryBlockType[]> {
```

Reply:
We have a similar pattern in other places. I think the root cause is in the component. We get the `libraryId` from `useParams`, and we can't type guard it so that it is not undefined before calling the hooks (because we can't have conditional hook calls). Could you help us with that?

Reply:
I see. In that case, it's fine to leave as is :)